text/0001-package-structure.md

```diff
@@ -0,0 +1,135 @@
+- Feature Name: `package-structure`
+- Start Date: 2024-10-24
+- RFC PR: [CMIP-REF/rfcs#0001](https://github.com/CMIP-REF/rfcs/pull/0001)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes how we should structure the git repository to enable the rapid development of the project while it is immature.
+The core of the project will have some reuse outside of the REF frontend
+so it should be publishable as a separate package.
+The goal is to make it easier for users to extend the project with new metrics providers.
+
+# Motivation
+[motivation]: #motivation
+
+* At the start of this project we will be in a state of flux until we have a working prototype.
+ APIs will change, and we will be adding and removing features.
+* Each metrics package will come with its own set of dependencies. Odds are that they will clash with other packages.
+* We want something that is easy to extend in future, incorporating new metrics providers/metrics.
+* Consistent tooling across the project
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+It is proposed that we start with a mono-repo structure consisting of multiple packages,
+each with their own set of dependencies.
+The tooling and CI will be shared across all projects.
+There will be a `packages` directory at the root of the repo, with each package in its own subdirectory.
```

Comment:
So the package is one of benchmarking packages,  will its development here be synced back to its original external repo or vice verse?